tasks/broad/UltimaGenomicsGermlineFilteringThreshold.wdl

```diff
@@ -0,0 +1,388 @@
+            output_vcf_name = base_file_name + ".extracted.vcf.gz",
+            disk_size = medium_disk
+    }
+
+    call FilterSampleVCF{
+        input:
+            input_vcf = GatherVcfs.output_vcf
+    }
+
+    call FilterSymbolicAlleles{
+        input:
+            base_file_name = base_file_name,
+            input_vcf = FilterSampleVCF.output_vcf_file,
+            input_vcf_index = FilterSampleVCF.output_vcf_index
+         }
+
+    call CompareToGroundTruth{
+        input:
+            gtr_vcf = gtr_vcf,
+            gtr_vcf_index = gtr_vcf_index, 
+            gtr_highconf = gtr_highconf_intervals,
+            left_sample_name = sample_name_calls,
+            right_sample_name = sample_name_gtr,
+            input_vcf = FilterSymbolicAlleles.output_vcf,
+            input_vcf_index = FilterSymbolicAlleles.output_vcf_index,
+            input_vcf_name = base_file_name,
+            
+            ref_fasta = ref_fasta,
+            ref_dict = ref_dict,
+            ref_fasta_sdf = ref_fasta_sdf,
+            runs_file = runs_file,
+            annotation_intervals = annotation_intervals,
+
+            disk_size = 2*(ceil(size(input_vcf, "GB") + size(gtr_vcf, "GB") + size(ref_fasta, "GB") + size(ref_fasta_sdf, "GB")))+10,
+            flow_order = flow_order
+
+    }
+
+    call EvaluateResults {
+        input:
+          h5_input = CompareToGroundTruth.compare_h5,
+          input_vcf_name = base_file_name,
+          disk_size = 10,
+          score_key = score_key
+    }
+
+    scatter (idx in range(length(input_vcf))) {
+        call HardThresholdVCF {
+            input:
+              thresholds = EvaluateResults.thresholds_report,
+              input_vcf = AnnotateSampleVCF.output_vcf_file[idx],
+              input_vcf_index = AnnotateSampleVCF.output_vcf_index[idx],
+              score_key = score_key,
+              output_basename = base_file_name,
+              disk_size = 3*ceil(size(AnnotateSampleVCF.output_vcf_file[idx], "GB")) + 14
+        }
+    }
+
+    output {
+        Array[File] output_vcf = HardThresholdVCF.output_vcf
+        Array[File] output_vcf_index = HardThresholdVCF.output_vcf_index
+        File eval_report_h5 = EvaluateResults.eval_report_h5
+        File thresholds_report = EvaluateResults.thresholds_report
+    }
+}
+
+task ExtractSample {
+    input {
+        String sample_name
+        File input_vcf
+        File input_vcf_index
+        String docker = "gcr.io/terra-project-249020/jukebox_vc:test_jc_optimize_3d7509"
+    }
+    String output_vcf = basename(input_vcf, ".vcf.gz") + ".~{sample_name}.vcf.gz"
+    command <<<
+        set -eo pipefail
+        source ~/.bashrc
+        conda activate genomics.py3
+        bcftools view -s ~{sample_name} ~{input_vcf} -o  tmp.vcf.gz -O z 
+        bcftools index -t tmp.vcf.gz
+        bcftools view -h tmp.vcf.gz | sed 's/COMBINED_TREE_SCORE,Number=A,/COMBINED_TREE_SCORE,Number=\.,/g' > hdr.fixed.txt
```

Comment:
Parts of this WDL seem like they could be generically useful, no? It might be worth thinking about exposing arguments needed in this Ultima-specific use case and moving towards generic tasks/sub-WDLs at some point, if not now.

Reply:
Thank you for commenting this because I realized that this line is outdated and is no longer needed. I made the task more generic, but I'm not sure if you're suggesting that it live somewhere else more generic as well? Should I move it to Utilities.wdl? Every time I touch anything in Utilities I have to update the versions of every pipeline that uses them which seems like a minor drawback.